### ISSUE TITLE
create database statement parse failed

### ISSUE THREAD

**Issue by zr-hebo** [vitessio/vitess#3823]
## Description
Execute statement like 
>create database test_db;

will get error message:
>syntax error at position 16 near 'database'

This can have a better error tips, after parse statement, like "unsupport create database statement".

**Comment by LepX** [comment]
### Test
```golang
func TestParseCreateStatement(t *testing.T) {
	ddl := "CREATE DATABASE `sqloptimizer` /*!40100 DEFAULT CHARACTER SET latin1 */;"
	stmt, err := sqlparser.Parse(ddl)
	if err != nil {
		fmt.Println(err)
	}

	....
}
```

### Ooutput
```
syntax error at position 16 near 'database'
```

**Comment by sougou** [comment]
Closed via #3835
